src/main/java/asynchronous/typing/TypingTestTemplate.java

```diff
@@ -58,57 +58,47 @@ public TypingTestTemplate(MessageReceivedEvent event, String[] args)
 	
 	
 	/**
-	 * Checks whether a typing submission is valid, and calculates its statistics.
+	 * Checks whether a typing submission is valid, and replies in turn.
 	 * @param event
 	 */
 	@Override
 	public void onMessageReceived(MessageReceivedEvent event)
 	{
-		// Data
-		double timeTakenMillis = (System.currentTimeMillis()) - startTime;
-		double wordsPerMinute = (numChars / timeTakenMillis) * 12000;
-				
+		// Gather Data.
+		long timeTakenMillis = (System.currentTimeMillis()) - startTime;
 		long userID = event.getAuthor().getIdLong();
 		MessageChannel answerChannel = event.getChannel();
-
-		if (answerChannel.getIdLong() != channel.getIdLong()) {return;}
-		else if (submissions.getUserIDs().contains(userID)) {return;}
-		else if (event.getAuthor().isBot()) {return;}
-
-		// Definitions.
 		String userTypingSubmission = event.getMessage().getContentRaw();
 		String userTag = event.getAuthor().getAsTag();
-
 		
+		// Filter invalid messages.
+		if (answerChannel.getIdLong() != channel.getIdLong()) {return;}
+		else if (submissions.getUserIDs().contains(userID)) {return;}
+		else if (event.getAuthor().isBot()) {return;}
 		
-		int editDistance = new LevenshteinDistance().apply(prompt, userTypingSubmission);
-		double accuracy = 100* (double)(prompt.length() - editDistance) / (double)prompt.length();
-
+		// Calculation.
```

Comment:
honestly this is not at all a big deal, but I feel these comments might be a bit redundant, because I can already see that the code is doing a calculation. Generally comments are better if they explain why a piece of code is doing something rather than what it's doing. good vid on this topic >> https://www.youtube.com/watch?v=k5DJxz6_xMA

but not a big enough deal to change so no worries

Reply:
You're right. I think we (mostly me) need to go through what we've written, but for the time being, I think this is alright.